ui/server/src/test/scala/pl/touk/nussknacker/ui/api/helpers/TestFactory.scala

```diff
@@ -138,14 +138,18 @@ object TestFactory extends TestPermissions{
     override def findJobStatus(name: ProcessName): Future[Option[ProcessState]] =
       Future.successful(managerProcessState.get())
 
-    override def deploy(processId: ProcessVersion, deploymentData: DeploymentData,
-                        processDeploymentData: ProcessDeploymentData, savepoint: Option[String]): Future[Option[ExternalDeploymentId]] =
+    override def deploy(processVersion: ProcessVersion, deploymentData: DeploymentData,
+                        processDeploymentData: ProcessDeploymentData, savepoint: Option[String]): Future[Option[ExternalDeploymentId]] = {
+      deploys.updateAndGet(l => processVersion::l)
       deployResult
+    }
 
     private var deployResult: Future[Option[ExternalDeploymentId]] = Future.successful(None)
 
     private val managerProcessState = new AtomicReference[Option[ProcessState]](prepareProcessState(defaultProcessStateStatus))
 
+    val deploys = new AtomicReference[List[ProcessVersion]](Nil)
```

Comment:
Why not `ConcurrentSkipListSet`?

Reply:
true, in the end I replaced with `ConcurrentLinkedQueue`